k8s/client.go

```diff
@@ -16,27 +25,79 @@ type Client interface {
 	WatchIngresses(Watcher) error
 }
 
-type noopClient struct {
+type impl struct {
```

Comment:
We don't have to... let's decide on a convention.
We could also call it after the interface, but lowercase maybe? So 'client'.